platform/wayland/cog-platform-wl.c

```diff
@@ -85,6 +85,51 @@ struct _CogWlPlatformClass {
 struct _CogWlPlatform {
     CogPlatform parent;
 };
+static void resize_to_largest_output();
+static void resize_window();
+static void configure_surface_geometry();
+static void xdg_surface_on_configure(void *data, struct xdg_surface *surface, uint32_t serial);
+static void xdg_toplevel_on_configure(void *               data,
+                                      struct xdg_toplevel *toplevel,
+                                      int32_t              width,
+                                      int32_t              height,
+                                      struct wl_array *    states);
+static void xdg_toplevel_on_close(void *data, struct xdg_toplevel *xdg_toplevel);
+static void shell_surface_ping(void *data, struct wl_shell_surface *shell_surface, uint32_t serial);
```

Comment:
Having the xdg and shell surface/popup listener declarations mixed together like this is a little confusing, better to keep them together and separated by newlines to make reading easier. I would also consider putting them closer to where they're used, rather than just at the very top like this.